src/providers/MeetingProvider/MeetingManager.ts

```diff
@@ -457,12 +477,55 @@ export class MeetingManager implements AudioVideoObserver {
         this.selectAudioInputDeviceError = error;
         console.error('Failed to choose audio input device.', error);
       }
-      this.selectedAudioInputDevice = deviceId;
+
+      let innerDevice = null;
+      if (isAudioTransformDevice(device)) {
+        innerDevice = await device.intrinsicDevice();
+      } else {
+        innerDevice = device;
+      }
+
+      if (innerDeviceId) {
```

Comment:
In what case, this `if` block's `else will be executed? if `innerDeviceId` is some valid value then wont it set the `this.selectedAudioInputDevice` and no check rest of the else-if ?

Reply:
Sorry I got confused on the `innerDeviceId` and `innerDevice`.